src/AppBundle/Service/ReportService.php

```diff
@@ -41,59 +42,60 @@ public function findById($id)
 
     /**
      * Set report type based on CasRec record (if existing)
-     * If not found, sets a 102 report (to discuss if an execption works better here)
      *
-     * @param Report $report
+     * @param Client $report
+     *
+     * @return string|null type
      */
-    public function setReportTypeBasedOnCasrec(Report $report)
+    public function getReportTypeBasedOnCasrec(Client $client)
     {
-        $casRec = $this->casRecRepository->findOneBy(['caseNumber' => $report->getClient()->getCaseNumber()]);
+        $casRec = $this->casRecRepository->findOneBy(['caseNumber' => $client->getCaseNumber()]);
         if ($casRec instanceof CasRec) {
-            $report->setType(CasRec::getTypeBasedOnTypeofRepAndCorref($casRec->getTypeOfReport(), $casRec->getCorref()));
-            return;
-        }
-
-        if($report instanceof Report and $report->getType() !== null)
-        {
-            $report->setType($report->getType());
-            return;
+            return CasRec::getTypeBasedOnTypeofRepAndCorref($casRec->getTypeOfReport(), $casRec->getCorref());
         }
 
-        $report->setType(Report::TYPE_102);
+        return null;
     }
 
     /**
      * Create new year's report copying data over (and set start/endDate accordingly).
      *
-     * @param Report $report
+     * @param Report $oldReport
      *
      * @return Report
      */
-    public function createNextYearReport(Report $report)
+    public function createNextYearReport(Report $oldReport)
     {
-        //lets clone the report
-        $client = $report->getClient();
-        $newReport = new Report($client);
-        $newReport->setType($report->getType());
-        $this->setReportTypeBasedOnCasrec($newReport);
-
-        $newReport->setStartDate($report->getEndDate()->modify('+1 day'));
-        $newReport->setEndDate($report->getEndDate()->modify('+12 months -1 day'));
-        $newReport->setReportSeen(false);
-        $newReport->setNoAssetToAdd($report->getNoAssetToAdd());
-
-        // clone assets
-        foreach ($report->getAssets() as $asset) {
+        if (!$oldReport->getSubmitted()) {
+            throw new \RuntimeException("Can't create a new year report based on an unsubmitted report");
+        }
+
+        $client = $oldReport->getClient();
+
+        $startDate = clone $oldReport->getEndDate();
```

Comment:
it needs a clone, as modify acts on the object, and if $oldReport is flushed by mistake, the old report date will be altered